torch/nn/modules/container.py

```diff
@@ -457,85 +451,81 @@ def __getitem__(self: T, idx: slice) -> T:
 
     def __getitem__(self, idx):
         if isinstance(idx, slice):
-            return self.__class__(list(self._parameters.values())[idx])
+            start, stop, step = idx.indices(len(self))
+            out = self.__class__()
+            for i in range(start, stop, step):
+                out.append(self[i])
```

Comment:
Just checking my understanding: this creates a new `ParameterList` with the elements of the slice, purposefully routing through the `append()` logic?

Reply:
Correct